apps/wallet/src/ui/app/hooks/useGetTransactionsByAddress.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiAddress, type SuiTransactionResponse } from '@mysten/sui.js';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+
+import { api } from '_redux/store/thunk-extras';
+const STALE_TIME = 1000 * 2;
+
+async function getTransactionsByAddress(
+    address: string,
+    cursor?: string
+): Promise<SuiTransactionResponse[]> {
+    const rpc = api.instance.fullNode;
+    const txnsIds = await api.instance.fullNode.getTransactions(
+        {
+            FromAddress: address,
+        },
+        cursor
+    );
+    return rpc.getTransactionWithEffectsBatch(txnsIds.data);
+}
+
+// Fetch transactions on mount and every 2 seconds
+export function useGetTransactionsByAddress(
+    address?: SuiAddress | null
+): UseQueryResult<SuiTransactionResponse[], unknown> {
+    return useQuery(
+        ['transactions-by-address', address],
+        async () => {
+            if (!address) {
+                throw new Error('No wallet address provided');
```

Comment:
Technically impossible so you could also just non-null assert but I'm also totally fine keeping this here.